Devscord.DiscordFramework/Services/HelpDataCollector.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Autofac;
+using Devscord.DiscordFramework.Commons.Extensions;
+using Devscord.DiscordFramework.Framework.Architecture.Controllers;
+using Devscord.DiscordFramework.Services.Factories;
+using Devscord.DiscordFramework.Services.Models;
+
+namespace Devscord.DiscordFramework.Services
+{
+    public class HelpDataCollector : IService
+    {
+        private readonly IComponentContext _componentContext;
+
+        public HelpDataCollector(IComponentContext componentContext)
+        {
+            this._componentContext = componentContext;
+        }
+
+        public IEnumerable<CommandInfo> GetCommandsInfo(Assembly botAssembly)
+        {
+            var controllers = botAssembly.GetTypesByInterface<IController>();
+            var commandFactory = new CommandsInfoFactory();
```

Comment:
CommandsInfoFactory to powinien być field private readonly
ponieważ CommandsInfoFactory nie trzyma żadnej konfiguracji specyficznej dla konkretnego zapytania
a jeśli możemy ograniczyć ilość instancji w aplikacji => róbmy to
tym bardziej że w tym przypadku przez to zyskamy na czytelności